internal/transport/transport_test.go

```diff
@@ -2746,3 +2756,85 @@ func (s) TestClientSendsAGoAwayFrame(t *testing.T) {
 		t.Errorf("Context timed out")
 	}
 }
+
+// hangingConn is a net.Conn wrapper for testing, simulating hanging connections
+// after a GOAWAY frame is sent, of which Write operations pause until explicitly signaled
+// or a timeout occurs.
+type hangingConn struct {
+	net.Conn
+}
+
+func (hc *hangingConn) Read(b []byte) (n int, err error) {
+	n, err = hc.Conn.Read(b)
+	return n, err
+}
+
+func (hc *hangingConn) Write(b []byte) (n int, err error) {
+	n, err = hc.Conn.Write(b)
+	if isGreetingsDone.Load() == true {
+		// Add a delay which is more than goAwayLoopyWriterTimeout
+		time.Sleep(2 * time.Second)
+	}
+	return n, err
+}
+
+func (hc *hangingConn) Close() error {
+	return hc.Conn.Close()
+}
+
+func (hc *hangingConn) LocalAddr() net.Addr {
```

Comment:
you don't need to override all methods. I think only write and close are probably enough for the test case

Reply:
Yes please delete all these methods that just call the underlying `net.Conn` directly.